substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -2588,5 +2604,40 @@ pub mod pallet {
 
 			Ok(Pays::No.into())
 		}
+
+		/// Permissionlessly fixes the over-stake.
+		///
+		/// If a user has lower fund than their stake in the ledger, this function will fix the
+		/// over-stake to their actual staked funds.
+		///
+		/// This might be caused because of slashing of funds by another pallet.
+		#[pallet::call_index(32)]
+		#[pallet::weight(T::DbWeight::get().reads_writes(2, 1))]
+		pub fn withdraw_overstake(
+			origin: OriginFor<T>,
+			stash: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+
+			let ledger = Self::ledger(Stash(stash.clone()))?;
+			let actual_stake = asset::staked::<T>(&stash);
+			let force_withdraw_amount = ledger.total.defensive_saturating_sub(actual_stake);
+
+			// ensure there is something to force unstake.
+			ensure!(!force_withdraw_amount.is_zero(), Error::<T>::BoundNotMet);
+
+			// we ignore if active is 0. It implies the locked amount is not actively staked. The
+			// account can still get away from potential slash, but we can't do much better here.
+			StakingLedger {
+				total: actual_stake,
+				active: ledger.active.saturating_sub(force_withdraw_amount),
+				..ledger
+			}
+			.update()?;
+
+			Self::deposit_event(Event::<T>::Withdrawn { stash, amount: force_withdraw_amount });
+
+			Ok(Pays::No.into())
```

Comment:
```suggestion
			Ok(())
```

Just worried that since this should really not be needed unless if there is a bug, making it feeless might create more issues such as someone rinsing and repeating this..